packages/lib/src/index.ts

```diff
@@ -80,14 +73,29 @@ export function isMediaOutput(output: Output): output is MediaOutput {
   return "uri" in output
 }
 
+export function isMediaHosted(output: Output): output is MediaOutput {
+  return "url" in output && output.url !== null
+}
+
+
 export type InferredOutput<TInput> = TInput extends MessagesInput
   ? MessageOutput
   : TInput extends PromptInput
   ? TextOutput | MediaOutput
   : Output
 
+
+// Base set of options for all requests.
+export interface Options<TModel, TInput extends Input = Input> {
+  // Identifier of the model to use. Defaults to the user's current model, but can be overridden here.
+  model?: TModel
+  // How many completion choices to attempt to generate. Defaults to 1. If the
+  // model doesn't support more than one, then an array with a single element will be returned.
+  numOutputs?: number
+}
+
 // CompletionOptions allows you to specify options for the completion request.
-export interface CompletionOptions<TModel, TInput extends Input = Input> {
+export interface CompletionOptions<TModel, TInput extends Input = Input> extends Options<TModel, TInput> {
```

Comment:
i think this line could now be `
export interface CompletionOptions<TModel> extends Options<TModel>` ?